pkg/upgrade/upgrade.go

```diff
@@ -13,12 +14,28 @@ import (
 // ManagedInstances finds all the Jaeger instances for the current operator and upgrades them, if necessary
 func ManagedInstances(c client.Client) error {
 	list := &v1.JaegerList{}
-	opts := &client.ListOptions{}
+	opts := client.MatchingLabels(map[string]string{
+		"app.kubernetes.io/instance":   viper.GetString(v1.ConfigIdentity),
+		"app.kubernetes.io/managed-by": "jaeger-operator",
```

Comment:
Confused - this seems to better reflect the description for this k8s label, but above this label is used to store the actual identity of the jaeger operator instance? So does it need a non-k8s label to define the jaeger operator instance identity?

Reply:
Sorry,  this is wrong -- a previous iteration had the identity set to "instance", but I later changed to use `managed-by`. I'll change this code.